astropy/samp/tests/test_client.py

```diff
@@ -35,7 +35,8 @@ def samp_hub(request):
     """A fixture that can be used by client tests that require a HUB."""
     my_hub = SAMPHubServer()
     my_hub.start()
-    request.addfinalizer(my_hub.stop)
+    yield my_hub
```

Comment:
This fixture didn't return `my_hub`, so it shouldn't yield it either.
```suggestion
    yield
```